LibertyDSNP/sdk-ts
# Features/announcement module 177120830

Problem
=======
As a user of the SDK I want to be able to announce a stored batch.
[link to Pivotal Tracker #177120830](https://www.pivotaltracker.com/story/show/177120830)

Solution
========
We created a  new Module named Announce

with @rosalinekarr 

Change summary:
---------------
* Create an announcement module that successfully calls `batch` function on the `Announce` contract 
* Add integration tests
* Add documentation under `/docs/modules`
* Add utility called `activityPubHash` to ensure ActivityPub objects are ordered properly

Steps to Verify:
----------------
1. Spin up docker-node-poa
1. Deploy `Announce` contract from contracts repo to the docker-node-poa. 
1.  Retrieve contract address for `Announce` contract that was deployed from`Contracts` repo and set it as **BATCH_CONTRACT_ADDRESS** Env variable in .env file
1. In `.env` - Set RPC_URL - to the address a node on your  docker-node-poa network is running on
1. In `.env` - Set TESTING_PRIVATE_KEY to the private key of an account that exists on your docker-node-poa that has an account balance
1. run `npm run tests`
1. Integration test should run that successfully calls `batch` function on announce contract and receives correct result


TODO:
-------------
1. Ci is not yet configured and integration tests will not run on CI until [this story](https://www.pivotaltracker.com/story/show/177584391) is completed
